app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,60 @@
+class TripsController < ApplicationController
+
+  def index
+    @trips = Trip.all
+  end
+
+  def edit
+    @trip = Trip.find_by(id: params[:id])
+  end
+
+  def update
+    @trip = Trip.find(params[:id])
+    if @trip.update(trip_params)
+      redirect_to root_path
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    trip = Trip.find_by(id: params[:id])
+    trip.destroy
+    redirect_to root_path
+  end
+
+  def create
+    if params[:passenger_id]
+      passenger = Passenger.find_by(id:
+        params[:passenger_id])
+        driver = Driver.all.sample
+        @trip = passenger.trips.new(driver_id: driver.id, passenger_id: passenger.id, date: "2018-06-01", rating: nil, cost: nil)
+        if @trip.save
+          redirect_to root_path
+        else
+          render :edit
```

Comment:
If a trip is not successfully created, _should_ the user get redirected to the `trip edit` page?